manual-tests/automated-test.py

```diff
@@ -0,0 +1,342 @@
+import argparse
+import fnmatch
+import os
+import platform
+import shutil
+import signal
+import time
+
+import psutil
+import yaml
+from yaml.loader import SafeLoader
+
+
+class ZF_conf:
+    def __init__(self):
+        self.zenohd_path = None
+        self.zf_plugin_path = None
+        self.zf_router_path = None
+        self.zenohd_process = None
+        self.zfctl_path = None
+        self.zfctl_json_path = None
+        self.zf_deamon_yaml = None
+        self.zenoh_flow_daemon = None
+        self.uuid_runtimes = None
+        self.libzenoh_plugin = None
+
+
+zf_conf = ZF_conf()
+
+zf_command = {
+    "zenohd_cmd": None,
+    "runtimes_cmd": None,
+    "dataflow_yaml_cmd": None,
+    "instances_cmd": None,
+}
+instance = {
+    "uuid_instances": None,
+    "flow": None,
+    "operators": None,
+    "sinks": None,
+    "sources": None,
+    "connectors": None,
+    "links": None,
+}
+
+
+def find_dir(name, path):
+    for root, dirs, files in os.walk(path):
+        if name in dirs:
+            return os.path.join(root, name)
+
+
+def find(pattern, path):
+    result = []
+    for root, dirs, files in os.walk(path):
+        for name in files:
+            if fnmatch.fnmatch(name, pattern):
+                result.append(os.path.join(root, name))
+    return result
+
+
+pid_zenohd = 0
+json_flag = False
+
+zenohflow_release_flag = False
+
+parser = argparse.ArgumentParser(
+    description="Script to automatically test that zenoh-flow works"
+)
+
+parser.add_argument(
+    "-b",
+    "--build",
+    nargs="?",
+    type=str,
+    default="False",
+    help="Specifies a different directory for Zenoh-Flow artifacts (default=debug)",
+)
+parser.add_argument(
+    "-z",
+    "--zenohd",
+    nargs="?",
+    type=str,
+    default="/.local/bin/zenohd",
+    help="Specifies a different path for the zenohd process (default=/.local/bin/zenohd)",
+)
+parser.add_argument(
+    "-c",
+    "--zfctl",
+    nargs="?",
+    type=str,
+    default="/.config/zenoh-flow/zfctl.json",
+    help="Specifies a different path for the zenoctl process (default=/.config/zenoh-flow/zfctl.json)",
+)
+parser.add_argument(
+    "-p",
+    "--plugin",
+    nargs="?",
+    type=str,
+    default="/.config/zenoh-flow/zenoh-zf-plugin-01.json",
+    help="Specifies a different path for Zenoh-Flow plugin file (default=/.config/zenoh-flow/zenoh-zf-plugin-01.json)",
+)
+args = parser.parse_args()
+
+home_path = os.path.expanduser("~")
+zenoh_release_flag = args.build
+
+if (
+    args.zenohd == "/.local/bin/zenohd"
+    or args.zfctl == "/.config/zenoh-flow/zfctl.json"
+):
+    print(home_path)
+    zf_conf.zenohd_path = home_path + str(args.zenohd)
+    zf_conf.zfctl_path = home_path + str(args.zfctl)
+    zf_conf.zf_plugin_path = home_path + str(args.plugin)
+else:
+    zf_conf.zenohd_path = args.zenohd
+    zf_conf.zfctl_path = args.zfctl
+    zf_conf.zf_plugin_path = args.plugin
```

Comment:
It is better to put the correct path in the default options. If users run the "help" option and see `"/.config/…"` they can assume that something is wrong.

There is the function: `os.path.expanduser()`.

What I suggest you do: provide a default option with `~` in the path & call `expanduser` in all cases (be it the default option or provided by the user).